### ISSUE TITLE
Exception on non-BMP character input in Windows Terminal

### ISSUE THREAD

**Issue by rmccampbell** [prompt-toolkit/python-prompt-toolkit#1477]
When attempting to input a non-BMP character, like 😀, in the (new) Windows Terminal, the following exception is thrown and the input fails:

```
Unhandled exception in event loop:
  File "c:\users\ryan\appdata\local\programs\python\python39\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "c:\users\ryan\appdata\local\programs\python\python39\lib\site-packages\prompt_toolkit\input\win32.py", line 512, in ready
    callback()
  File "c:\users\ryan\appdata\local\programs\python\python39\lib\site-packages\prompt_toolkit\application\application.py", line 662, in read_from_input
    keys = self.input.read_keys()
  File "c:\users\ryan\appdata\local\programs\python\python39\lib\site-packages\prompt_toolkit\input\win32.py", line 75, in read_keys
    return list(self.console_input_reader.read())
  File "c:\users\ryan\appdata\local\programs\python\python39\lib\site-packages\prompt_toolkit\input\win32.py", line 233, in read
    all_keys = list(self._get_keys(read, input_records))
  File "c:\users\ryan\appdata\local\programs\python\python39\lib\site-packages\prompt_toolkit\input\win32.py", line 292, in _get_keys
    for key_press in self._event_to_key_presses(ev):
  File "c:\users\ryan\appdata\local\programs\python\python39\lib\site-packages\prompt_toolkit\input\win32.py", line 330, in _event_to_key_presses
    ascii_char = u_char.encode("utf-8")

Exception 'utf-8' codec can't encode character '\ud83d' in position 0: surrogates not allowed
```
It seems that the terminal is sending the character as separate utf-16 surrogate codes, which breaks the win32 input logic. Either the two key events would somehow have to be merged into one KeyPress with the full character or the unicode would have to be normalized somehow later on.

**Comment by rmccampbell** [comment]
Also it seems that non-BMP input doesn't work in the normal Conhost, but there is no exception, it simply does nothing. Not sure if this is a separate problem which may be harder to debug.

**Comment by rmccampbell** [comment]
Fixed by #1482
